pkg/kncloudevents/http_client_new.go

```diff
@@ -0,0 +1,167 @@
+/*
+Copyright 2023 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package kncloudevents
+
+import (
+	"crypto/tls"
+	"crypto/x509"
+	"fmt"
+	nethttp "net/http"
+	"sync"
+	"time"
+
+	"go.opencensus.io/plugin/ochttp"
+	duckv1 "knative.dev/pkg/apis/duck/v1"
+	"knative.dev/pkg/tracing/propagation/tracecontextb3"
+)
+
+const (
+	defaultRetryWaitMin = 1 * time.Second
+	defaultRetryWaitMax = 30 * time.Second
+)
+
+var (
+	clientsMutex   sync.Mutex
+	clients        map[string]*nethttp.Client
+	connectionArgs *ConnectionArgs
+)
+
+func init() {
+	clients = make(map[string]*nethttp.Client)
+}
+
+func getClientForAddressable(addressable duckv1.Addressable) (*nethttp.Client, error) {
+	clientsMutex.Lock()
+	defer clientsMutex.Unlock()
+
+	clientKey := addressable.URL.String()
+
+	client, ok := clients[clientKey]
+	if !ok {
+		newClient, err := createNewClient(addressable)
+		if err != nil {
+			return nil, fmt.Errorf("failed to create new client for addressable: %w", err)
+		}
+
+		clients[clientKey] = newClient
+
+		client = newClient
+	}
+
+	return client, nil
+}
+
+func createNewClient(addressable duckv1.Addressable) (*nethttp.Client, error) {
+	var base = nethttp.DefaultTransport.(*nethttp.Transport).Clone()
+
+	if addressable.CACerts != nil && *addressable.CACerts != "" {
+		certPool, err := x509.SystemCertPool()
+		if err != nil {
+			return nil, fmt.Errorf("could not get system cert pool: %w", err)
+		}
+
+		if ok := certPool.AppendCertsFromPEM([]byte(*addressable.CACerts)); !ok {
+			return nil, fmt.Errorf("failed to append CA certs from PEM to cert pool")
+		}
+
+		base.TLSClientConfig = &tls.Config{
+			MinVersion: tls.VersionTLS12,
+			RootCAs:    certPool,
+		}
+	}
```

Comment:
Can we use the eventing TLS library?

Reply:
I wanted, but then we have a circular dependency, because eventingtls uses kncloudevents too :/

Reply:
Then why not using eventingtls package similar to "ServerManager" we could have a "ClientManager"?

Reply:
I removed the dependency in #6980. Not sure about having a "ClientManager" in eventingtls, as not all clients need a TLS config.

Reply:
ServerManager manages Server that aren't necessarily using TLS too .. isn't that the same? I'm not saying one way or the other just making sure we have a reasonable API

Reply:
Sure. The server manager always starts both servers (http & https) IIRC. And keeping it in the `kncloudevents` package and having the `getClientXYZ` private should encapsulate it from other use anyhow. So users should only use the `kncloudevents.CloudEventRequest.Send|SendWithRetries()` methods. 
But whatever fits best...